MatrixSDK/Data/RoomList/Common/MXRoomListDataFilterable.swift

```diff
@@ -0,0 +1,134 @@
+                                                subpredicates: [predicate1, predicate2])
+            predicates.append(predicate)
+        }
+        
+        if !filterOptions.onlySuggested {
+            //  data types
+            if !filterOptions.dataTypes.isEmpty {
+                let predicate = NSPredicate(format: "(%K & %d) != 0",
+                                            #keyPath(MXRoomSummaryProtocol.dataTypes),
+                                            filterOptions.dataTypes.rawValue)
+                predicates.append(predicate)
+            }
+            
+            //  not data types
+            if !filterOptions.notDataTypes.isEmpty {
+                let predicate = NSPredicate(format: "(%K & %d) == 0",
+                                            #keyPath(MXRoomSummaryProtocol.dataTypes),
+                                            filterOptions.notDataTypes.rawValue)
+                predicates.append(predicate)
+            }
+            
+            //  space
+            if let space = filterOptions.space {
+                //  specific space
+                let predicate = NSPredicate(format: "%@ IN %K", space.spaceId,
+                                            #keyPath(MXRoomSummaryProtocol.parentSpaceIds))
+                predicates.append(predicate)
+            } else {
+                //  home space
+                
+                // In case of home space we show a room if one of the following conditions is true:
+                // - Show All Rooms is enabled
+                // - It's a direct room
+                // - The room is a favourite
+                // - The room is orphaned
+                
+                let predicate1 = NSPredicate(value: filterOptions.showAllRoomsInHomeSpace)
+                
+                let directDataTypes: MXRoomSummaryDataTypes = .direct
+                let predicate2 = NSPredicate(format: "(%K & %d) != 0",
+                                             #keyPath(MXRoomSummaryProtocol.dataTypes),
+                                             directDataTypes.rawValue)
+                
+                let favoritedDataTypes: MXRoomSummaryDataTypes = .favorited
+                let predicate3 = NSPredicate(format: "(%K & %d) != 0",
+                                             #keyPath(MXRoomSummaryProtocol.dataTypes),
+                                             favoritedDataTypes.rawValue)
+                
+                let predicate4_1 = NSPredicate(format: "%K == NULL",
+                                               #keyPath(MXRoomSummaryProtocol.parentSpaceIds))
+                let predicate4_2 = NSPredicate(format: "%K.@count == 0",
+                                               #keyPath(MXRoomSummaryProtocol.parentSpaceIds))
+                let predicate4 = NSCompoundPredicate(type: .or,
+                                                     subpredicates: [predicate4_1, predicate4_2])
```

Comment:
When Steve and I were looking at filtering for the hotfix, we wondered if these predicate names could be made more descriptive to make it easier for someone looking at this with new eyes.

Reply:
Also, just leaving a note here about the tweaks made to filtering in the hotfix :)